src/index.ts

```diff
@@ -15,60 +13,90 @@ import {
   TwoFAVerificationStatus,
 } from './models/2fa-models';
 
+interface IInfobipAuthObject {
+  baseUrl: string;
+  authType: AuthType;
+  apiKey?: string;
+  username?: string;
+  password?: string;
+  oauthToken?: string;
+  ibssoToken?: string;
+}
 class Infobip {
   /**
    *
-   * @param {InfobipAuth} config - Configuration object for Infobip API
+   * @param { InfobipAuth | Http } config - Configuration object for Infobip API
    *
    */
 
   credentials: InfobipAuth;
   service: any;
   channels: any;
-  auth: any;
+  auth: Auth;
+  http: Http;
+  api: API;
+
+  constructor(InfobipObject: IInfobipAuthObject | Http) {
+    this.credentials = {
+      baseUrl: '',
+    };
+    this.http = new Http('');
+    if (InfobipObject instanceof Http) {
+      this.http = InfobipObject;
+      this.credentials = new InfobipAuth({
+        baseUrl: this.http.baseUrl,
+      });
+    } else {
+      const {
+        baseUrl,
+        authType,
+        apiKey,
+        username,
+        password,
+        oauthToken,
+        ibssoToken,
+      } = InfobipObject;
+      Validator.required(baseUrl, 'Infobip.baseUrl');
+      Validator.required(authType, 'Infobip.authType');
 
-  constructor({
-    baseUrl,
-    authType,
-    apiKey,
-    username,
-    password,
-    oauthToken,
-    ibssoToken,
-  }: InfobipAuth) {
-    Validator.required(baseUrl, 'Infobip.baseUrl');
-    Validator.required(authType, 'Infobip.authType');
+      password && Validator.string(password, 'Infobip.password');
+      username && Validator.string(username, 'Infobip.username');
+      apiKey && Validator.string(apiKey, 'Infobip.apiKey');
+      oauthToken && Validator.string(oauthToken, 'Infobip.oauthToken');
+      ibssoToken && Validator.string(ibssoToken, 'Infobip.ibssoToken');
 
-    password && Validator.string(password, 'Infobip.password');
-    username && Validator.string(username, 'Infobip.username');
-    apiKey && Validator.string(apiKey, 'Infobip.apiKey');
-    oauthToken && Validator.string(oauthToken, 'Infobip.oauthToken');
-    ibssoToken && Validator.string(ibssoToken, 'Infobip.ibssoToken');
+      this.credentials = new InfobipAuth({
+        baseUrl,
+        authType,
+        apiKey,
+        username,
+        password,
+        oauthToken,
+        ibssoToken,
+      });
+      this.http = new Http(
+        this.credentials.baseUrl,
+        this.credentials.authorization
+      );
+    }
 
-    this.credentials = new InfobipAuth({
-      baseUrl,
-      authType,
-      apiKey,
-      username,
-      password,
-      oauthToken,
-      ibssoToken,
-    });
+    this.api = new API(this);
```

Comment:
Acronym HTTP is Http, and API is API, maybe all acronyms should be only first capital letter?